inputgraph/inputgraph_test.go

```diff
@@ -0,0 +1,107 @@
+package inputgraph
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/earthly/earthly/conslogging"
+	"github.com/earthly/earthly/domain"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHashTargetWithDocker(t *testing.T) {
+	r := require.New(t)
+	target := domain.Target{
+		LocalPath: "./testdata/with-docker",
+		Target:    "with-docker-load",
+	}
+
+	ctx := context.Background()
+	cons := conslogging.New(os.Stderr, &sync.Mutex{}, conslogging.NoColor, 0, conslogging.Info)
+
+	org, project, hash, err := HashTarget(ctx, target, cons)
+	r.NoError(err)
+	r.Equal("earthly-technologies", org)
+	r.Equal("core", project)
+
+	hex := fmt.Sprintf("%x", hash)
+	r.Equal("9d2903bc18c99831f4a299090abaf94d25d89321", hex)
+
+	path := "./testdata/with-docker/Earthfile"
+	err = replaceInFile(path, "saved:latest", "other:latest")
```

Comment:
can we make use of a temp directory instead? manipulating the file directly could lead to a difficult to debug case where the final restore doesn't get run due to an error or seg fault.

furthermore, it could be odd to have your text editor prompt you to reload `./testdata/with-docker/Earthfile"` if you happen to have it open.